bin/find-spark-home.cmd

```diff
@@ -0,0 +1,44 @@
+@echo off
+
+rem
+rem Licensed to the Apache Software Foundation (ASF) under one or more
+rem contributor license agreements.  See the NOTICE file distributed with
+rem this work for additional information regarding copyright ownership.
+rem The ASF licenses this file to You under the Apache License, Version 2.0
+rem (the "License"); you may not use this file except in compliance with
+rem the License.  You may obtain a copy of the License at
+rem
+rem    http://www.apache.org/licenses/LICENSE-2.0
+rem
+rem Unless required by applicable law or agreed to in writing, software
+rem distributed under the License is distributed on an "AS IS" BASIS,
+rem WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+rem See the License for the specific language governing permissions and
+rem limitations under the License.
+rem
+
+rem Path to Python script finding SPARK_HOME
+set FIND_SPARK_HOME_SCRIPT=%~dp0find_spark_home.py
+
+rem Default to standard python interpreter unless told otherwise
+set PYTHON_RUNNER=python
+if not "x%PYSPARK_DRIVER_PYTHON%" =="x" (
+  set PYTHON_RUNNER=%PYSPARK_DRIVER_PYTHON%
+)
+
+rem Only attempt to find SPARK_HOME if it is not set.
+if "x%SPARK_HOME%"=="x" (
+  rem We are pip installed, use the Python script to resolve a reasonable SPARK_HOME
+  if exist "%FIND_SPARK_HOME_SCRIPT%" (
+    rem If there is no python installed it will fail with message:
+    rem 'python' is not recognized as an internal or external command,
+    for /f "delims=" %%i in ('%PYTHON_RUNNER% %FIND_SPARK_HOME_SCRIPT%') do set SPARK_HOME=%%i
```

Comment:
Hm.. actually, looks we deal with `PYSPARK_PYTHON`:

https://github.com/apache/spark/blob/a36a76ac43c36a3b897a748bd9f138b629dbc684/bin/find-spark-home#L38

```bash
$ PYSPARK_PYTHON=python3
$ echo ${PYSPARK_PYTHON:-"python"}
python3
```

```bash
$ echo ${PYSPARK_PYTHON:-"python"}
python
```